graylog2-web-interface/src/views/components/sidebar/fields/ChangeMode.jsx

```diff
@@ -0,0 +1,26 @@
+// @flow strict
+import * as React from 'react';
+
+type Props = {
+  changeMode: (newMode: string) => void,
+  currentMode: string,
+  mode: string,
+  text: string,
+  title: string,
+};
+
+const ChangeMode = ({ changeMode, currentMode, mode, text, title }: Props) => {
```

Comment:
`ChangeMode` is not an ideal name, it is not really giving away what it is doing. Maybe something like `FieldGroup`?
I think we can also align props naming with React standards. Instead of `changeMode` using `onSelect` or `onClick` would give away what it's doing, instead of `currentMode`, we can use a boolean `selected` prop, allowing the user of the component to determine if it should be shown as selected or not.

Reply:
Alright, I only see one disadvantage with the select prop, we need to define the group (previously mode) twice for each `FieldGroup`. For the `group` and the `select` prop.

Reply:
Agreed, it's not ideal.